tests/test_nihao_handler.py

```diff
@@ -0,0 +1,144 @@
+    return {
+        'fields': {
+            'stars': {
+                'age': 'age',
+                'mass': 'mass',
+                'metallicity': 'metallicity',
+                'coords': 'pos',
+                'velocity': 'vel'
+            },
+            'gas': {
+                'density': 'density',
+                'temperature': 'temperature'
+            },
+            'dm': {
+                'mass': 'mass'
+            }
+        },
+        'particles': {
+            'stars': {
+                'coords': 'kpc',
+                'mass': 'Msun',
+                'age': 'Gyr',
+                'velocity': 'km/s',
+                'metallicity': 'dimensionless'
+            },
+            'gas': {
+                'density': 'Msun/kpc^3',
+                'temperature': 'K'
+            },
+            'dm': {
+                'mass': 'Msun'
+            }
+        },
+        'units': {
+            'stars': {
+                'coords': 'kpc',
+                'mass': 'Msun',
+                'age': 'Gyr',
+                'velocity': 'km/s',
+                'metallicity': 'dimensionless'
+            },
+            'gas': {
+                'density': 'Msun/kpc^3',
+                'temperature': 'K'
+            },
+            'dm': {
+                'mass': 'Msun'
+            },
+            'galaxy': {
+                'redshift': 'dimensionless',
+                'center': 'kpc',
+                'halfmassrad_stars': 'kpc'
+            }
+        },
+        'galaxy': {
+            'redshift': 0.1,
+            'halfmassrad_stars': 5.0
+        }
+    }
+
+@pytest.fixture
+def handler_with_mock_data(mock_simulation, mock_config):
+    """Fixture to initialize the NihaoHandler with mocked data."""
+    mock_simulation.stars.get.return_value = np.array([0.1, 0.2, 0.3])
+    mock_simulation.stars["pos"] = np.array([[0.0, 0.0, 0.0], [1.0, 1.0, 1.0], [2.0, 2.0, 2.0]]) 
+    mock_simulation.stars["mass"] = np.array([1.0, 2.0, 3.0]) 
+
+    mock_simulation.gas.loadable_keys.return_value = ["density", "temperature"] 
+    mock_simulation.gas["density"] = np.array([1.0, 2.0, 3.0]) 
+    mock_simulation.gas["temperature"] = np.array([100.0, 200.0, 300.0]) 
+
+    import pynbody
+    pynbody.load = MagicMock(return_value=mock_simulation)
+
+    handler = NihaoHandler(path="mock_path", halo_path="mock_halo_path", config=mock_config)
+    handler.sim = mock_simulation
+    return handler
+
+def test_nihao_handler_initialization(handler_with_mock_data):
+    """Test initialization of NihaoHandler."""
+    assert handler_with_mock_data is not None
+
+def test_load_data(handler_with_mock_data):
+    """Test if data is loaded correctly."""
+    handler_with_mock_data.load_data()
+    assert handler_with_mock_data.center is not None
+
+def test_get_galaxy_data(handler_with_mock_data):
+    """Test retrieval of galaxy data."""
+    galaxy_data = handler_with_mock_data.get_galaxy_data()
+    assert galaxy_data is not None
```

Comment:
As you specify in the beginning, what the galaxy data are, you could check here not only that they are not None, you could also check that they have the correct value